python/cuml/decomposition/tsvd_mg.pyx

```diff
@@ -23,570 +22,548 @@ import ctypes
-        self.components_ptr = None
-        self.explained_variance_ptr = None
-        self.explained_variance_ratio_ptr = None
-        self.singular_values_ptr = None
-
-        self.algo_dict = {'full': COV_EIG_DQ,
-                          'auto': COV_EIG_DQ,
-                          'jacobi': COV_EIG_JACOBI}
-
-    def _get_algorithm_c_name(self, algorithm):
-        return self.algo_dict[algorithm]
-
-    def _initialize_arrays(self, n_components, n_rows, n_cols):
-
-        self.trans_input_ = zeros(n_rows*n_components, self.dtype)
-
-        self.components_ = zeros(n_cols*n_components, self.dtype)
-
-        self.explained_variance_ = \
-            cudf.Series(zeros(n_components, self.dtype))
-
-        self.explained_variance_ratio_ = \
-            cudf.Series(zeros(n_components, self.dtype))
-
-        self.mean_ = cudf.Series(zeros(n_cols, self.dtype))
+                                bool verbose) except +
 
-        self.singular_values_ = \
-            cudf.Series(zeros(n_components, self.dtype))
 
-        self.noise_variance_ = cudf.Series(zeros(1, dtype=self.dtype))
+class TSVDMG(TruncatedSVD):
 
-    def _get_ctype_ptr(self, obj):
-        # The manner to access the pointers in the gdf's might change, so
-        # encapsulating access in the following 3 methods. They might also be
-        # part of future gdf versions.
-        return obj.device_ctypes_pointer.value
+    def __init__(self, **kwargs):
+        super(TSVDMG, self).__init__(**kwargs)
 
-    def _get_column_ptr(self, obj):
-        return self._get_ctype_ptr(obj._column._data.to_gpu_array())
-
-    def _get_gdf_as_matrix_ptr(self, gdf):
-        return self._get_ctype_ptr(gdf.as_gpu_matrix())
-
-    def _fit_spmg(self, X, _transform=True, gpu_ids=[]):
-
-        if (not isinstance(X, np.ndarray)):
-            msg = "X matrix must be a Numpy ndarray. Dask will be supported" \
-                  + " in the next version."
-            raise TypeError(msg)
-
-        n_gpus = len(gpu_ids)
-
-        n_rows = X.shape[0]
-        n_cols = X.shape[1]
-        self.params.n_rows = n_rows
-        self.params.n_cols = n_cols
+    def _initialize_arrays(self, n_components, n_rows, n_cols):
 
-        if (n_rows < 2):
-            msg = "X must have at least two rows."
-            raise TypeError(msg)
+        self.components_ary = rmm.to_device(zeros(n_components*n_cols,
+                                                  dtype=self.dtype))
+
+        self.explained_variance_ = cudf.Series(zeros(n_components,
+                                               dtype=self.dtype))
+
+        self.explained_variance_ratio_ = cudf.Series(zeros(n_components,
+                                                     dtype=self.dtype))
+
+        self.singular_values_ = cudf.Series(zeros(n_components,
+                                                  dtype=self.dtype))
+
+    def _build_dataFloat(self, arr_interfaces):
+        cdef floatData_t ** dataF = < floatData_t ** > \
```

Comment:
```suggestion
        cdef floatData_t** dataF = <floatData_t**> \
```

why the extra spaces for the pointers? Not a huge deal, and not worth delaying for that, but if you have the chance I'd love to see the pointer spacing more normal :)

Reply:
Removed.